libs/core/src/lib/shellbar/shellbar.component.ts

```diff
@@ -26,45 +31,154 @@ export type ShellbarSizes = 's' | 'm' | 'l' | 'xl';
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ShellbarComponent implements AfterContentInit {
+export class ShellbarComponent implements AfterContentInit, AfterViewInit {
     /** Size of Shellbar component 's' | 'm' | 'l' | 'xl' */
     @Input()
     size: ShellbarSizes = 'm';
 
+    @Input()
+    responsive = false;
+
+    @Input()
+    sizesWidth = {
+        s: 320,
+        m: 720,
+        l: 1024,
+        xl: 1400
+    };
+
     /**
      * Whether the Shellbar is used with Side Navigation
      * When set to true, the responsive paddings are not applied
      */
     @Input()
     sideNav = false;
 
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
+
     /** @hidden */
     @ContentChild(ComboboxComponent, { static: false })
     comboboxComponent: ComboboxComponent;
 
+    /** @hidden */
+    @ContentChild(SelectComponent, { static: false })
+    selectComponent: SelectComponent;
+
     /** @hidden */
     @ContentChildren(forwardRef(() => ButtonComponent))
     buttons: QueryList<ButtonComponent>;
 
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef, private _cdr: ChangeDetectorRef) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._attachResizeListener();
+    }
+
     /** @hidden */
     ngAfterContentInit(): void {
-        this.applyShellbarModeToCombobox();
-        this.applyShellbarModeToButtons();
+        this._applyShellbarModeToButtons();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._onResize();
+        this._applyShellbarModeToCombobox();
+        this._applyShellbarModeToSelect();
+    }
+
+    /** to apply view change on select menu open */
+    handleSelectIsOpenChange(): void {
+        this._subscriptions.add(
+            this.selectComponent.isOpenChange.subscribe((isOpen) => {
+                if (isOpen) {
+                    setTimeout(() => {
+                        this.selectComponent._controlElementRef.nativeElement.classList.remove('is-expanded');
```

Comment:
could you instead of manipulating native element classes add input properties to the select\combobox? i.e. `inputClass`, `buttonClass`, etc

The same for methods, instead of focusing by `nativeElement.focus()` just add `focus` method to combobox\select, so in case the inner structure of those components got changed you don't have to change everything outside of them.

Reply:
Inputs would be exposed to the app developers, not sure that's what we want, unless you mean not inputs but just hidden properties on select/combobox